### ISSUE TITLE
Class path resources have invalid content length

### ISSUE THREAD

**Issue by dsyer** [oracle/graal#1683]
```java
public class Resources {

	public static void main(String[] args) throws Exception {
		new Resources().run();
	}

	private void run() throws Exception {
		System.err.println(Resources.class.getResource("Resources.class").openConnection().getContentLength());
	}

}
```

Compile it and then run it:

```
$ native-image -H:+ReportUnsupportedElementsAtRuntime -Dfile.encoding=UTF-8 -H:Name=resources -H:IncludeResources=Resources.class -cp target/classes/ Resources
...
$ ./resources 
-1
$ java -cp target/classes/ Resources
927
```

**Comment by olpaw** [comment]
Thanks for the nice reproducer. I'll have a look.

**Comment by olpaw** [comment]
Fixed on master 36e9f9b7f9f8f845ad3d729eff2c43d2356a9739
